src/main/java/Duke.java

```diff
@@ -1,10 +1,162 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    /**
+     * Number of tasks scheduled
+     */
+    public static int taskCounter = 0;
+
+    /**
+     * Task type 100 objects to store the tasks the user will create
+     */
+    public static Task[] scheduledTasks = new Task[100];
+
+    /**
+     * This is the main function which is responsible for executing all the functions
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        greet();
+        runDuke();
+        greetBye();
+    }
+
+    /**
+     * Greets the user by printing some introductory messages
+     */
+    private static void greet() {
+        printLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printLine();
+    }
+
+    /**
+     * Prints a line on the screen
+     */
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Takes in input from the user and executes the given instructions.
+     * If user input is "list", it calls the list() function to list all the tasks.
+     * If user input is done x, where x is a valid task number, it calls the MarkTaskAsDone() function to mark the task as done.
+     * If user has scheduled a task by writing a statement beginning with "event", "deadline" or "todo",
+     * then it adds the task in the tasks list by calling the addTaskToList() function.
+     *
+     * @param userInput            userInput stores the input String entered by the user.
+     * @param taskCompletionStatus taskCompletionStatus Stores the status of the task, true if completed, false otherwise.
+     */
+    private static void runDuke() {
+        int i;
```

Comment:
Looks like this local variable `i` is never used in this function, perhaps you could consider removing it.

Reply:
Sure I will do it. Thanks for notifying.